src/lavinmq/mqtt/broker.cr

```diff
@@ -75,8 +75,7 @@ module LavinMQ
       end
 
       def subscribe(client, topics)
-        session = sessions[client.client_id]? || sessions.declare(client)
-        session.client = client
```

Comment:
Why does this reset the session and remove the subscriptions?

Reply:
because when we trigger the client= setter, it removes the consumer on the session (we only have one consumer in mqtt, the old client) and replace it with the new one. But if you have a client with clean_session set to true, it will delete the session when the consumer is removed (it inherits the auto_delete functionality from Queue)

Reply:
oh i see

Reply:
That should probably be refactored too, probably as a part of #957

Reply:
Yes definitely, probably time for that soon.. this solves the bug in the mean time though